openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -13,272 +13,263 @@
+        # get_next_query_params function.
+        self.offset = None
+
+    @staticmethod
+    def _compute_date_range(date_start: str, days: int) -> str:
+        """
+        Given an ISO formatted date string and a number of days, compute the
+        ISO string that represents the start date plus the days provided.
+        """
+        date_end = date.fromisoformat(date_start) + timedelta(days=days)
+        return date_end.isoformat()
+
+    def get_media_type(self, record: dict) -> str:
+        return constants.IMAGE
+
+    @property
+    def endpoint(self) -> str:
+        """
+        If this is being run as a dated DAG, **only one request is ever issued** to
+        retrieve all updated IDs. As such, the dated version will only return one
+        endpoint. The full run DAG does require the typical offset + limit, which gets
+        recomputed on each call to this property.
+        """
+        list_endpoint = f"{self.base_endpoint}/list"
+        if self.date:
+            end_date = self._compute_date_range(self.date, self.default_process_days)
+            # Get a list of objects uploaded/updated within a date range
+            # http://phylopic.org/api/#method-image-time-range
+            endpoint = f"{list_endpoint}/modified/{self.date}/{end_date}"
+        else:
+            # Get all images and limit the results for each request.
+            endpoint = f"{list_endpoint}/{self.offset}/{self.batch_limit}"
+        logger.info(f"Constructed endpoint: {endpoint}")
+        return endpoint
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        """
+        Since the query range is determined via endpoint, this only increments the range
+        to query.
+        """
+        if self.offset is None:
+            self.offset = 0
+        else:
+            self.offset += self.batch_limit
+        return {}
+
+    def get_should_continue(self, response_json):
+        """
+        Override for upstream "return True". Dated runs will only ever make 1 query so
+        they should not continue to loop.
+        """
+        return not bool(self.date)
+
+    @staticmethod
+    def _get_response_data(response_json) -> dict | list | None:
+        """
+        Intermediate method for pulling out results from a Phylopic API request.
+        """
+        if response_json and response_json.get("success") is True:
+            return response_json.get("result")
+
+    def get_batch_data(self, response_json):
+        """
+        Process the returned IDs.
+
+        The Phylopic API returns only lists of IDs in the initial request. We must take
+        this request and iterate through all the IDs to get the metadata for each one.
+        """
+        data = self._get_response_data(response_json)
+
+        if not data:
+            logger.warning("No content available!")
+            return None
+
+        return data
+
+    @staticmethod
+    def _image_url(_uuid: str) -> str:
```

Comment:
Is `uuid` reserved? Why does it have a leading `_`? I don't _see_ it imported in this module, but maybe it's being defined globally somehow? Is this an Airflow specific thing?

Reply:
That _is_ strange. Looks like this comes from the original code and also applied to some other variables like `_endpoint`, which definitely shouldn't be reserved. Maybe meant to denote an internal variable, but it's not done consistently 🤔

I don't see any reason we couldn't omit the leading `_`, we have variables named `uuid` elsewhere in provider scripts. Probably a leftover artifact of refactoring multiple times!

Reply:
Oop, definitely holdover from the original code. I think it's more convention than anything, since `uuid` _is_ a module and it might be ideal to make the distinction between what this is and that module. That said, we aren't using that module here as y'all point out so I'll change it 🙂